pkg/gitreceive/build.go

```diff
@@ -190,7 +190,21 @@ func build(conf *Config, s3Client *s3.S3, kubeClient *client.Client, builderKey,
 	if err != nil {
 		return fmt.Errorf("fetching builder logs (%s)", err)
 	}
-	log.Debug("size of logs streamed %v", size)
+	log.Debug("size of streamed logs %v", size)
+
+	// check the state and exit code of the build pod.
+	// if the code is not 0 return error
+	if err := waitForPodEnd(kubeClient, newPod.Namespace, newPod.Name, conf.BuilderPodTickDuration(), conf.BuilderPodWaitDuration()); err != nil {
+		return fmt.Errorf("error getting builder pod status (%s)", err)
+	}
+	buildPod, err := kubeClient.Pods(newPod.Namespace).Get(newPod.Name)
+	if err != nil {
+		return fmt.Errorf("error getting builder pod status (%s)", err)
+	}
+	state := buildPod.Status.ContainerStatuses[0].State.Terminated
```

Comment:
@bacongobbler I didn't understand your comment. ContainerStatuses[0] is because the builder pod only contains 1 container. I can change that to avoid a future update to this code.